File: module-1/12_Polymorphism/student-exercise/java/src/main/java/com/techelevator/CreditCardAccount.java
package com.techelevator;

public class CreditCardAccount implements Accountable {
	private String accountHolder;
	private String accountNumber;
	private int debt;
	
	public CreditCardAccount(String accountHolder, String accountNumber) {
		super();
		this.accountHolder = accountHolder;
		this.accountNumber = accountNumber;
		this.debt = 0;
	}
	
	public String getAccountHolder() {
		return accountHolder;
	}

	public String getAccountNumber() {
		return accountNumber;
	}

	public int getDebt() {
		return debt;
	}

	public int pay(int amountToPay) {
		debt -= amountToPay;
		return debt;
	}
	
	public int charge(int amountToCharge) {
		debt += amountToCharge;
		return debt;
	}

	@Override
	public int getBalance() {
		
		return -1 * debt;
	}
	
	
}

File: module-1/12_Polymorphism/exercise-final/dotnet/BankTellerExercise/CreditCardAccount.cs
namespace BankTellerExercise
{
    public class CreditCardAccount : IAccountable
    {
        public int Debt { get; private set; }
        public string AccountHolderName { get; private set; }
        public string AccountNumber { get; private set; }
        public int Balance
        {
            get { return -Debt; }
        }

        public CreditCardAccount(string accountHolder, string accountNumber)
        {
            AccountHolderName = accountHolder;
            AccountNumber = accountNumber;
        }

        public int Pay(int amountToPay)
        {
            Debt -= amountToPay;
            return Debt;
        }

        public int Charge(int amountToCharge)
        {
            Debt += amountToCharge;
            return Debt;
        }

    }
}
